miner/worker.go

```diff
@@ -162,10 +162,20 @@ func (w *worker) commitNewWork() (*types.Block, error) {
 		return nil, fmt.Errorf("failed to prepare header for mining: %w", err)
 	}
 
+	// Instantiate stateDB and kick off prefetcher.
 	env, err := w.createCurrentEnvironment(parent, header, tstart)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create new current environment: %w", err)
 	}
+
+	// Ensure we always stop prefetcher after block building is complete.
+	defer func() {
+		if env.state == nil {
```

Comment:
I assume state can be nil here so `createCurrentEnvironment` can panic.

Reply:
Again, this logic mirrors `go-ethereum`: https://github.com/ethereum/go-ethereum/blob/460cc1673e583512cd83843ac2ab0186f0ddf1ab/miner/worker.go#L125-L127C3

Reply:
I tried to place this snippet as close as I could to where it is in `go-ethereum`: https://github.com/ethereum/go-ethereum/blob/460cc1673e583512cd83843ac2ab0186f0ddf1ab/miner/worker.go#L1021